alica_tests/src/conditions/conditions.cpp

```diff
@@ -222,4 +222,130 @@ bool TestHasNoError(const Blackboard* input, const RunningPlan* rp, const Blackb
     LockedBlackboardRO globalBlackboard(*gb);
     return !globalBlackboard.get<std::optional<std::string>>("testError").has_value();
 }
+bool GlobalCounterEqualTo(const Blackboard* input, const RunningPlan* rp, const Blackboard* gb)
+{
+    LockedBlackboardRO globalBlackboard(*gb);
+    LockedBlackboardRO localBlackboard(*input);
+    return globalBlackboard.hasValue("counter") && globalBlackboard.get<int64_t>("counter") == localBlackboard.get<int64_t>("value");
+}
+bool GlobalCounterLessThan(const Blackboard* input, const RunningPlan* rp, const Blackboard* gb)
+{
+    LockedBlackboardRO globalBlackboard(*gb);
+    LockedBlackboardRO localBlackboard(*input);
+    return globalBlackboard.hasValue("counter") && globalBlackboard.get<int64_t>("counter") < localBlackboard.get<int64_t>("value");
+}
+bool PlanAState2PlanBState(const Blackboard* input, const RunningPlan* rp, const Blackboard* gb)
+{
+    LockedBlackboardRO globalBlackboard(*gb);
+    std::string execOrder = globalBlackboard.get<std::string>("execOrder");
+    std::string suffix = "BehAAA::Init\n";
+    return globalBlackboard.get<int64_t>("counter") < 20 && execOrder.size() >= suffix.size() &&
+           0 == execOrder.compare(execOrder.size() - suffix.size(), suffix.size(), suffix);
+}
+bool PlanBState2PlanAState(const Blackboard* input, const RunningPlan* rp, const Blackboard* gb)
+{
+    LockedBlackboardRO globalBlackboard(*gb);
+    std::string execOrder = globalBlackboard.get<std::string>("execOrder");
+    std::string suffix = "BehBAA::Init\n";
+    return globalBlackboard.get<int64_t>("counter") < 20 && execOrder.size() >= suffix.size() &&
```

Comment:
20 is value mapped in globalCounterEqualTo, but here (& in other conditions) it is hard coded. Should use either value mapping for all or hardcoded for all.